### DISCUSSION TITLE
Plot no of active slip systems in IPF

### DISCUSSION THREAD

**Discussion by SagarChandra** [mtex-toolbox/mtex#1498] [Ask Anything] [answered]
Dear All,

I have a list of 100 tensile loading directions (Miller indices) and I want to plot the no. of active slip systems for each loading direction in the standard IPF plot for the FCC material. For instance, for loading along 110 direction, there are 4 active slip systems, 8 for 100 direction and so on. So, how do we plot the contours of no of active slip systems in the IPF for an FCC crystal.

Thanks and Regards,
Sagar

**Comment by kilir** [comment, marked as answer]
Hi,
in the broadest sense:
```
plot(h,data,'contour','fundamentalRegion')
```
Also have a look at this list, it's asked quite frequently.

Hope this helps.

Cheers,
Rüdiger

**Reply by SagarChandra** [reply]
Hi Rüdiger,

Thanks for the reply. Based on the available scripts shipped with MTeX and some MATLAB commands, I figured out that following lines are producing such a contour plot. Pasting them here for anyone in need:
```matlab
cs = crystalSymmetry('cubic',[3.615,3.615,3.615],'mineral','Copper');
sS = slipSystem.fcc(cs);
sSAll = sS.symmetrise('antipodal');
tol = 1e-1;                                          % Define a tolerance for distinguishing active/in-active system

r = plotS2Grid(fundamentalSector(cs,'antipodal'),'resolution',0.05*degree);
c = sSAll.SchmidFactor(r);                  % Calculate Schmidt factor
c(abs(c)<tol) = 0;                                 % Set Schmidt factor to zero if it's absolute value is less than the tolerance.
NoOfPotSlipSys = sum(c ~= 0, 2);       % Count no. of non-zero elements for quantifying no. of potential slip systems for a given loading                          direction
contourf(r,NoOfPotSlipSys,vector3d.Z,'smooth');                                       % Plot the contour of the scalar values.
```
PS: Please feel free to comment in case any irregularity is observed.

**Reply by ralfHielscher** [reply]
Hi Sagar,

would you mind write some lines of explanation to your code (what is depicted, why it is useful, how tol is set). 
We could then include this into our documentation.

Thank you very much :)

Ralf.

**Reply by SagarChandra** [reply]
Hi Ralf,

Thank you for your positive response ! Here is the script with explanation:-

%% This script counts the number of potential slip systems for a given loading direction and plots the scalar values as contour plots in the %% standard stereographic triangle. Given that plastic deformation in crystalline materials is primarily governed by activation of distinct slip %% systems, crystal orientation can have a significant influence on the total no. of slip systems and the ensuing quantifiable attributes like %% total accumulated plastic strain, dislocation density accumulation etc. Comparison of the number of potential slip systems and the    %% attributes of plastic deformation for a given loading direction can provide mechanistic rationales for the observed variations. 

%% The script runs fine with MATLAB 2020a and MTeX 5.5.2.

%First, define the crystal symmetry. Here, FCC Cu is assumed as the model material.
cs = crystalSymmetry('cubic',[3.615,3.615,3.615],'mineral','Copper'); 

% Define the slip systems as an MTEX variable
sS = slipSystem.fcc(cs); 

% Symmetrize them to have distinct set of systems. 
sSAll = sS.symmetrise('antipodal');

% While calculating Schmidt factor, it is possible (based on the MATLAB version one is using) that due to precision, numbers like 1.3e-16 or %-0.0029 appear after the calculation of Schmidt factor for a particular loading direction. To counter this, a tolerance value is defined so that %MATLAB ignores values less than the tolerance.  This tolerance should be checked based on the crystal structure of the material.
tol = 1e-1;                                          % Define a tolerance for distinguishing active/in-active system

% Define some directions
r = plotS2Grid(fundamentalSector(cs,'antipodal'),'resolution',0.05*degree); 

% Calculate Schmidt factor
c = sSAll.SchmidFactor(r); 
                 
% Set Schmidt factor to zero if it's absolute value is less than the tolerance.
c(abs(c)<tol) = 0;     

 % Count no. of non-zero elements for quantifying no. of potential slip systems for a given loading  direction                            
NoOfPotSlipSys = sum(c ~= 0, 2);      

%Plot the scalar values as a function of loading direction
contourf(r,NoOfPotSlipSys,vector3d.Z,'smooth');     

% Now, for any loading direction of interest, it can be checked as to how many potential systems will be activated by annotating them on the % created  contour plot.

hold on

m =[Miller(1,0,0,cs),
Miller(18,1,0,cs),
Miller(10,1,0,cs), ]

annotate(m,'label',char(m,'cell','latex'),'backgroundColor','white');


Thanks and Regards,
Sagar
